pkg/controller/job/job_controller_test.go

```diff
@@ -3140,11 +3230,7 @@ func TestEnsureJobConditions(t *testing.T) {
 			if len(gotList) != len(tc.expectList) {
 				t.Errorf("got a list of length %d, want %d", len(gotList), len(tc.expectList))
 			}
-			for i := range gotList {
-				// Make timestamps the same before comparing the two lists.
-				gotList[i].LastProbeTime = tc.expectList[i].LastProbeTime
-				gotList[i].LastTransitionTime = tc.expectList[i].LastTransitionTime
-			}
+			makeActualTimestampsAsExpected(gotList, tc.expectList)
```

Comment:
this looks like a hack :)

instead, you can ignore the field when comparing using `cmp.Diff` below.

See cmpopts https://pkg.go.dev/github.com/google/go-cmp/cmp/cmpopts#IgnoreFields

Reply:
Indeed, improved the comparison in the last commit